examples/custom_role/main.tf

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/******************************************
+  Provider configuration
+ *****************************************/
+provider "google" {
+  version = "~> 2.7"
+}
+
+provider "google-beta" {
+  version = "~> 2.7"
+}
+
+locals {
+  role_permissions = ["iam.roles.list", "iam.roles.create", "iam.roles.delete"]
+}
+
+resource "random_id" "rand_custom_id" {
+  byte_length = 2
+}
+
+/******************************************
+  Module custom_role call
+ *****************************************/
+module "custom-roles" {
+  source = "../../modules/custom_role_iam/"
+
+  role_level  = "project"
+  org_id      = var.org_id
```

Comment:
org_id shouldn't be necessary for role_level project right? Or am I missing something?

Reply:
org_id is a variable in modules/custom_role_iam and is required in this example.

Reply:
I can think of two ways. The reason I think this is worth doing is because as a user I would rather not pass in something like org_id="" so that I can instantiate this submod for a project. 

1) set default to "" for both and in the description add required if project/organization respectively. [This would be similar to how we handle the region var in GKE where its required if zonal.](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/ec864f18734b7eef375a6544da755aecd418ad71/variables.tf#L41)
2) abstract both to a single var say, var.id and make that required. User will pass in either org or poject id to this var. Depending on role_level at any point only one of the two resources would be instantiated. 

I will defer to @morgante for his thoughts if there are better approaches or if as-is is the way to go.

Reply:
I think a combination of `target_level` and `target_id` should work.

Reply:
Agreed, we should *not* require `org_id` for project-level roles.

Reply:
Done.